lib/dogma/rule/taken_name.ex

```diff
@@ -0,0 +1,86 @@
+defmodule Dogma.Rule.TakenName do
+  @moduledoc """
+  A rule that disallows function or macro names which overrides standard lib.
+
+  For example, this is considered valid:
+
+    def just_do_it do
+    end
+
+    defmacro make_your_dreams_come_true(clause, expression) do
+    end
+
+  While this is considered invalid:
+
+    def unless do
+    end
+
+    defmacro require(clause, expression) do
+    end
+  """
+
+  @behaviour Dogma.Rule
+  alias Dogma.Script
+  alias Dogma.Error
+
+  reserved_words = ~w(alias bc case cond exit function if import inbits inlist
+               is_atom is_binary is_bitstring is_boolean is_exception is_float
+               is_function is_integer is_list is_number is_pid is_port
+               is_record is_reference is_tuple lc quote raise receive require
+               respawn super throw try unless unless unquote use
+               )
```

Comment:
Perhaps we could grab all the function and macro names from Kernel and SpecialForms?